File: src/templates/java-compat-fc/src/newarch/java/com/TempName/TempNameViewManager.java
package com.TempName;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.module.annotations.ReactModule;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.facebook.react.uimanager.ViewManagerDelegate;
import com.facebook.react.uimanager.annotations.ReactProp;
import com.facebook.react.viewmanagers.TempNameManagerInterface;
import com.facebook.react.viewmanagers.TempNameManagerDelegate;

import com.TempName.*;

@ReactModule(name = TempNameViewManagerImpl.NAME)
class TempNameViewManager extends SimpleViewManager<TempName>
        implements TempNameManagerInterface<TempName> {

    private final ViewManagerDelegate<TempName> mDelegate;

    public TempNameViewManager(ReactApplicationContext context) {
        mDelegate = new TempNameManagerDelegate<>(this);
    }

    @Nullable
    @Override
    protected ViewManagerDelegate<TempName> getDelegate() {
        return mDelegate;
    }

    @NonNull
    @Override
    public String getName() {
        return TempNameViewManagerImpl.NAME;
    }

    @NonNull
    @Override
    protected TempName createViewInstance(@NonNull ThemedReactContext context) {
        return TempNameViewManagerImpl.createViewInstance(context);
    }

    @Override
    @ReactProp(name = "text")
    public void setText(TempName view, @Nullable String text) {
        TempNameViewManagerImpl.setText(view, text);
    }
}

File: src/templates/kotlin-compat-fc/src/oldarch/java/com/TempName/TempNameViewManager.kt
package com.TempName

import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp
import com.facebook.react.bridge.ReactApplicationContext

import com.TempName.*

class TempNameViewManager(var context: ReactApplicationContext) : SimpleViewManager<TempName>() {

  override fun getName() = TempNameViewManagerImpl.NAME

  public override fun createViewInstance(context: ThemedReactContext): TempName = TempNameViewManagerImpl.createViewInstance(context)

  @ReactProp(name = "text")
  fun setFoo(view: TempName?, text: String?) {
    TempNameViewManagerImpl.setText(view, text)
  }
}